Strip protocol from model path

rlbaker opened the pull request:
Took a whack at fixing https://github.com/jmorganca/ollama/issues/371 and reorganized the switch logic slightly as well.

Wasn't sure if it was better to strip all protocols or just `https://`, so if you'd like the latter I can switch it to just a `strings.TrimPrefix`. Happy to back out the updated switch code as well, just figured I'd do it while I was in there.

Thanks for a great tool, loving it so far. Hope this is helpful.

jmorganca commented on server/modelpath.go:
```diff
@@ -23,33 +24,37 @@ const (
 		registry = slashParts[0]
 		namespace = slashParts[1]
-		repository = strings.Split(slashParts[2], ":")[0]
+		repository = slashParts[2]
```
Hello! Thanks for opening a PR 😊 

This is close, but I think you'll want to apply this to `registry` vs `repository`. I added an example here: https://github.com/jmorganca/ollama/issues/371

rlbaker commented:
@jmorganca Hey, thanks for following up!

Perhaps I'm misunderstanding the intent, but I believe this is functioning as you describe. Here's an example with some variations: https://goplay.tools/snippet/fu7DWcOnozH

I'm also happy to undo all the changes except the protocol regex splitting if it's unnecessary to bring in these other changes. Don't want to take up your time reviewing code not related to the original issue 😄 

Thanks again for taking a look!

rlbaker commented:
Added some tests as well, just to be on the safe side.

jmorganca commented:
Great! Getting close. I'm still seeing this error but lmk if I might have something wrong:

```
% ./ollama run https://ollama.ai/m/wb
pulling manifest
Error: pull model manifest: Get "https:///v2///manifests/": http: no Host in request URL
```

Also, this is more of a nice to have, but there is an `--insecure` flag that's required for `http://` just to make sure the user explicitly knows it's an insecure protocol. If a user specifies `http://ollama.ai/m/wb` for example, we should error unless `--insecure` is specified.

rlbaker commented:
I added support for the `--insecure` flag allowing the `http://` protocol, associated error messages, and updated tests.

Since the `run` command also supports arbitrary url's and pulls the image if not found, I added support for the `--insecure` flag to it as well. Let me know if it's behaving the way you expect!

Here's the resulting behavior:

```bash
$ ./ollama pull http://registry.ollama.ai/library/llama2:latest

Error: insecure protocol http
```

```bash
$ ./ollama pull http://registry.ollama.ai/library/llama2:latest

Error: insecure protocol http
```

```bash
./ollama pull file://registry.ollama.ai/library/llama2:latest --insecure
Error: invalid protocol scheme
```

```bash
./ollama pull http://registry.ollama.ai/library/llama2:latest --insecure
pulling manifest
```

jmorganca reviewed (approved):
Looks great! Awesome work
